packet_packer.go

```diff
@@ -146,6 +147,14 @@ func (p *packetPacker) PackPacket() (*packedPacket, error) {
 	if len(payloadFrames) == 1 && p.stopWaiting != nil {
 		return nil, nil
 	}
+	if (len(payloadFrames) == 1 && p.ackFrame != nil) || (len(payloadFrames) == 2 && p.ackFrame != nil && p.stopWaiting != nil) { // check if this packet only contains an ACK and / or STOP_WAITING
```

Comment:
This looks very fragile. We already have code to determine whether a frame is retransmittable, right? Can we introduce a `hasRetransmittableFrames([]frames.Frame) bool` function that iterates the slice here?

Reply:
Good idea. Actually, that function already exists: `ackhandler.HasRetransmittableFrames([]wire.Frame) bool`.